Signal/src/ViewControllers/SelectThreadViewController.m

```diff
@@ -0,0 +1,428 @@
+
+    [self addNotificationListeners];
+
+    [self updateTableContents];
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    [self.navigationController.navigationBar setTranslucent:NO];
+}
+
+- (void)addNotificationListeners
+{
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(blockedPhoneNumbersDidChange:)
+                                                 name:kNSNotificationName_BlockedPhoneNumbersDidChange
+                                               object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(signalRecipientsDidChange:)
+                                                 name:OWSContactsManagerSignalRecipientsDidChangeNotification
+                                               object:nil];
+}
+
+- (void)dealloc
+{
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+- (void)createViews
+{
+    // Table
+    _tableViewController = [OWSTableViewController new];
+    _tableViewController.delegate = self;
+    _tableViewController.contents = [OWSTableContents new];
+    [self.view addSubview:self.tableViewController.view];
+    [_tableViewController.view autoPinWidthToSuperview];
+    [_tableViewController.view autoPinToTopLayoutGuideOfViewController:self withInset:0];
+    [_tableViewController.view autoPinEdgeToSuperviewEdge:ALEdgeBottom];
+
+    // Search
+    UISearchBar *searchBar = [UISearchBar new];
+    _searchBar = searchBar;
+    searchBar.searchBarStyle = UISearchBarStyleProminent;
+    searchBar.delegate = self;
+    searchBar.placeholder = NSLocalizedString(@"SEARCH_BYNAMEORNUMBER_PLACEHOLDER_TEXT", @"");
+    [searchBar sizeToFit];
+    _tableViewController.tableView.tableHeaderView = searchBar;
+}
+
+#pragma mark - UISearchBarDelegate
+
+- (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText
+{
+    [self updateTableContents];
+}
+
+- (void)searchBarSearchButtonClicked:(UISearchBar *)searchBar
+{
+    [self updateTableContents];
+}
+
+- (void)searchBarCancelButtonClicked:(UISearchBar *)searchBar
+{
+    [self updateTableContents];
+}
+
+- (void)searchBarResultsListButtonClicked:(UISearchBar *)searchBar
+{
+    [self updateTableContents];
+}
+
+- (void)searchBar:(UISearchBar *)searchBar selectedScopeButtonIndexDidChange:(NSInteger)selectedScope
+{
+    [self updateTableContents];
+}
+
+#pragma mark - Actions
+
+- (void)updateTableContents
+{
+    __weak SelectThreadViewController *weakSelf = self;
+    OWSTableContents *contents = [OWSTableContents new];
+    OWSTableSection *section = [OWSTableSection new];
+
+    // Threads
+    for (TSThread *thread in [self filteredThreadsWithSearchText]) {
+        [section addItem:[OWSTableItem itemWithCustomCellBlock:^{
+            SelectThreadViewController *strongSelf = weakSelf;
+            if (!strongSelf) {
+                return (InboxTableViewCell *)nil;
```

Comment:
Why are we casting nil? 

If it's to clobber the "nonnull" constraint declared in:

```
 typedef UITableViewCell *_Nonnull (^OWSTableCustomCellBlock)();
```

Then I think we should just be honest with ourselves and change the UITableViewCell returned by OWSTableCustomCellBlock to be Nullable, since apparently it can be nil.